src/Sarif/Baseline/ResultMatching/DataStructures/MatchedResults.cs

```diff
@@ -160,5 +171,21 @@ public Result CalculateBasedlinedResult(DictionaryMergeBehavior propertyBagMerge
             }
             return result;
         }
+
+        public override string ToString()
+        {
+            if (PreviousResult != null && CurrentResult != null)
+            {
+                return $"= {PreviousResult}";
+            }
+            else if (PreviousResult == null)
+            {
+                return $"+ {CurrentResult}";
+            }
+            else
+            {
+                return $"- {PreviousResult}";
+            }
```

Comment:
>} [](start = 12, length = 1)

This suggests that the ctor should verify that at least one of `current` and `previous` is non-`null`.

Reply:
Good idea. Added.